### ISSUE TITLE
unbalanaced dataset

### ISSUE THREAD

**Issue by nmningmei** [dribbleanalytics/2019-draft-ml#1]
Cool analysis! Saw it from Reddit and decided to look through your code. Very impressive. But your data seems not balanced, but you did not deal with it by adding ["class_weight='balanced'"](https://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html) to your classifiers. Would you mind adding this parameter and make the cross-validation scores again? Also, for the random forest implementation, since you did not actually control for model complexity, it will be better to use [XGBClassifier](https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.XGBClassifier) because it allows you to add L1/L2 penalty to control for the model complexity. For the sake of model comparison, I will also suggest KNN and naive Bayes models from scitkitlearn to just test the non-linear assumption of your data.

**Comment by dribbleanalytics** [comment]
Thanks!

I actually wasn't familiar with the class_weight addition. Because I've already run the analysis and posted it, I will not change it for this specific project, as doing so would change results, accuracy metrics, etc. for something already published. However, I will definitely add it in future posts. Thanks for your suggestions about other models; I've used an XGBClassifier and KNN in previous posts, and was considering a naive Bayes model this time, but decided to go for the simpler linear models. I'll be sure to write about that next time.
